### PR TITLE
First draft for implementing a unified selectBone

### PR DESCRIPTION
**Pull request by phorward**
selectOneBone() refers to selectBone(), where selectMultiBone() refers to selectBone(multiple=True). This is much more convenient with the general ViUR development standards.

### CODE REVIEW

File: bones/selectBone.py [hunk 1]
```diff
@@ -50,77 +41,90 @@ def __init__(self, defaultValue=[], values={}, *args, **kwargs):
...
-			values = data[name]
-		else:
-			values = None
+	def fromClient(self, valuesCache, name, data):
+		values = data.get(name)
+
+		# single case
+		if not self.multiple:
+			for key in self.values.keys():
+				if str(key) == str(values):
```
Comment (tsteinruecken) [inline bones/selectBone.py]: We should probably change this to unicode()

File: bones/selectBone.py [hunk 2]
```diff
@@ -50,77 +41,90 @@ def __init__(self, defaultValue=[], values={}, *args, **kwargs):
...
+					err = self.isInvalid(key)
+					if not err:
+						valuesCache[name] = key
+
+					return err
+
+			return "No or invalid value selected"
+
+		# multiple case
```
Comment (tsteinruecken) [inline bones/selectBone.py]: Personally, i would prefer an clear else: case here for better readability

**Comment by phorward** [comment]
Both suggestions had been incorporated.

**Comment by tsteinruecken** [comment]
Haven't tested it yet in detail, but looks good to me so far. :+1:

**Pull request merged**